src/component/tooltip/TooltipHTMLContent.ts

```diff
@@ -60,17 +60,21 @@ function mirrorPos(pos: string): string {
 }
 
 function assembleArrow(
-    backgroundColor: ColorString,
+    tooltipModel: Model<TooltipOption>,
     borderColor: ZRColor,
     arrowPosition: TooltipOption['position']
 ) {
     if (!isString(arrowPosition) || arrowPosition === 'inside') {
         return '';
     }
 
+    const backgroundColor = tooltipModel.get('backgroundColor');
+    const borderWidth = tooltipModel.get('borderWidth');
+
     borderColor = convertToColorString(borderColor);
     const arrowPos = mirrorPos(arrowPosition);
-    let positionStyle = `${arrowPos}:-6px;`;
+    const arrowSize = borderWidth >= 5 ? borderWidth : 5;
+    let positionStyle = `${arrowPos}:-${Math.sqrt(2 * arrowSize * arrowSize) / 2 + borderWidth}px;`;
```

Comment:
The position of arrow seems not so accurated(shown as previous screenshot). We may need to minus half of the border width.

And the calculated number may have many decimal places, it would be better to fix it as an integer or remain two decimal places.

Reply:
You mean horizontal placement right?

Reply:
It seems both horizontal and vertical positions are off a bit. I think the position of arrow is correct(vertical middle / horizontal center of the tooltip container), but the position of the tooltip container is not correct, it should consider arrow size when transforming the container's x/y.

Reply:
@g7i Hi, any updates on this offset issue?

Reply:
I'm almost done with it. By tomorrow EOD will update the PR.
Thank you

Reply:
I'm almost done with it. Will update the PR by tomorrow EOD.
Thank you

Reply:
@g7i Great!

Reply:
Good job! We'll help review it ASAP. 👍